src/project.ts

```diff
@@ -152,3 +172,28 @@ export function path(
   }
   return "/" + result.join("/");
 }
+
+// We need a sha1 commit for the API call so we return early if rev is a sha1 and else we look into branches.
+// As fallback we go for the head
+// TODO: Add tags parsing
+export function parseRevision(head: string, revision: string, branches?: [string, string][]): string {
```

Comment:
It took me a while to understand what this function did, even after reading the comment.. Maybe calling it `getOid` would be more appropriate? Since we're not really parsing the revision.